File: iconics-java/src/main/java/com/open/iconics/animation/BlinkAlphaProcessor.java
package com.open.iconics.animation;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.text.TextPaint;

import androidx.annotation.IntRange;
import androidx.annotation.NonNull;

import com.open.iconics.IconicsBrush;

public class BlinkAlphaProcessor extends IconicsAnimationProcessor {
    /**
     * Duration used for all instances of this processor by default. 500 ms by default.
     */
    public static int defaultDuration = 500;

    @IntRange(from = 0, to = 255)
    private int mMinimumAlpha = 0;
    @IntRange(from = 0, to = 255)
    private int mMaximumAlpha = 255;

    {
        mRepeatMode = REVERSE;
        mDuration = defaultDuration;
    }

    /**
     * @param minimumAlpha The alpha which will be used as minimal available value.
     */
    @NonNull
    public BlinkAlphaProcessor minimumAlpha(@IntRange(from = 0, to = 255) int minimumAlpha) {
        mMinimumAlpha = minimumAlpha;
        return this;
    }

    /**
     * @param maximumAlpha The alpha which will be used as maximal available value.
     */
    @NonNull
    public BlinkAlphaProcessor maximumAlpha(@IntRange(from = 0, to = 255) int maximumAlpha) {
        mMaximumAlpha = maximumAlpha;
        return this;
    }

    /**
     * @return The minimal available alpha.
     */
    @IntRange(from = 0, to = 255)
    public int getMinimumAlpha() {
        return mMinimumAlpha;
    }

    /**
     * @return The maximal available alpha.
     */
    @IntRange(from = 0, to = 255)
    public int getMaximumAlpha() {
        return mMaximumAlpha;
    }

    @Override
    @NonNull
    public String animationTag() {
        return "blink_alpha";
    }

    @Override
    protected void processPreDraw(
            @NonNull Canvas canvas,
            @NonNull IconicsBrush<TextPaint> iconBrush,
            @NonNull IconicsBrush<Paint> iconContourBrush,
            @NonNull IconicsBrush<Paint> backgroundBrush,
            @NonNull IconicsBrush<Paint> backgroundContourBrush) {

        int alphaByPercent = (mMaximumAlpha - mMinimumAlpha) / 100;

        int alpha = (int) (getAnimatedPercent() * alphaByPercent);

        iconBrush.setAlpha(alpha);
        iconContourBrush.setAlpha(alpha);
        backgroundBrush.setAlpha(alpha);
        backgroundContourBrush.setAlpha(alpha);
    }
}

File: iconics-kotlin/src/main/java/com/open/iconics/animation/BlinkAlphaProcessor.kt
package com.open.iconics.animation

import android.animation.TimeInterpolator
import android.graphics.Canvas
import android.graphics.Paint
import android.text.TextPaint
import androidx.annotation.IntRange
import com.open.iconics.IconicsBrush
import com.open.iconics.animation.IconicsAnimationProcessor.RepeatMode.REVERSE

open class BlinkAlphaProcessor(
    /** The minimal available alpha. */
    @IntRange(from = 0, to = 255) open var minimumAlpha: Int = 0,
    /** The maximal available alpha. */
    @IntRange(from = 0, to = 255) open var maximumAlpha: Int = 255,

    override var interpolator: TimeInterpolator = DEFAULT_INTERPOLATOR,
    override var duration: Long = DEFAULT_DURATION,
    override var repeatCount: Int = INFINITE,
    override var repeatMode: RepeatMode = REVERSE,
    override var isStartImmediately: Boolean = true
) : IconicsAnimationProcessor(interpolator, duration, repeatCount, repeatMode, isStartImmediately) {

    companion object {
        /** Duration used for all instances of this processor by default. 500 ms by default. */
        @JvmField var DEFAULT_DURATION = 500L
    }

    override val animationTag: String = "blink_alpha"

    override fun processPreDraw(
        canvas: Canvas,
        iconBrush: IconicsBrush<TextPaint>,
        iconContourBrush: IconicsBrush<Paint>,
        backgroundBrush: IconicsBrush<Paint>,
        backgroundContourBrush: IconicsBrush<Paint>
    ) {

        val alphaByPercent = (maximumAlpha - minimumAlpha) / 100

        val alpha = (animatedPercent * alphaByPercent).toInt()

        iconBrush.alpha = alpha
        iconContourBrush.alpha = alpha
        backgroundBrush.alpha = alpha
        backgroundContourBrush.alpha = alpha
    }
}
